ledger_device_sdk/src/nbgl.rs

```diff
@@ -821,6 +865,116 @@ impl NbglGenericReview {
     }
 }
 
+pub struct NbglStreamingReview {
+    icon: nbgl_icon_details_t,
+    tx_type: TransactionType,
+}
+
+impl NbglStreamingReview {
+    pub fn new(tx_type: TransactionType) -> NbglStreamingReview {
```

Comment:
@yogh333 I have added the transaction type parameter to the `new` method. it's saved in the structure and used in various places (`start`, `continue_review` and `finish`)

I didn't want to have a separate function to set it.

What do you think ? Should I add this parameter to the other review structures as required ?

Reply:
Yes ok for me

Reply:
But I am wondering if it can be also generalised to other structs to provide constant strings when calling `new()`, e.g.:

- `NbglReview::new(title, subtitle, finish_title)`
- `NbglHomeAndSettings::new(info_contents)`
- `NbgAddressReview::new(verify_string)`

Reply:
I would prefer keeping the "builder pattern" style functions that seem to allow greater flexibility in using the structures of our API. Athough it's true that I'm not sure we would call `new` and then **not** call `titles` (for the `NbglReview`)